vtex-apps/b2b-quotes-graphql
# [B2BQUOTES-22] Send emails when quotes are created or updated

#### What problem is this solving?

This PR updates the app so that appropriate users are notified when quotes are created or updated. Specifically:

- when a quote is created (using the "submit to sales rep" button), all users with the Sales Admin role will receive an email
- when a quote is updated, all users who have interacted with the quote (i.e. all emails that appear in the quote's update history) will receive an email

The PR provides default email templates for the two types of emails (`QuoteCreated` and `QuoteUpdated`). These will be automatically added to the store's account if they don't exist when the following GraphQL operations are executed: `getQuote`, `getQuotes`, `createQuote`

#### How to test it?

New version linked here: https://quotes--sandboxusdev.myvtex.com

1. Go to https://quotes--sandboxusdev.myvtex.com/admin/customers
2. Search for your vtex.com.br email and assign yourself as the Organization Admin of Aaaa Organization (in any cost center)
3. Search for your vtex.com email and assign yourself as the Sales Admin (in any organization and cost center)
4. Go to https://quotes--sandboxusdev.myvtex.com and login as your vtex.com.br email
5. Add products to your cart and then go to /b2b-quotes/create
6. Add a name and description to the quote and click `Submit to Sales Rep`
7. You should receive an email at your vtex.com address notifying that the quote was created
8. Log out of the storefront and log back in as your vtex.com email
9. Go to /b2b-quotes
10. Edit the quote that you just created, changing whatever you want, and save it
11. Both your vtex.com and vtex.com.br emails should receive notification that the quote was updated
